app/controllers/rubygems_controller.rb

```diff
@@ -18,8 +18,13 @@ def index
   end
 
   def show
-    @latest_version = @rubygem.versions.most_recent
-    @versions       = @rubygem.public_versions(5)
+    if @rubygem
```

Comment:
I have added a new `before_action` to set blacklisted gem [here](https://github.com/rubygems/rubygems.org/pull/1433/commits/42ba3a6750c9d29e62e3d84e4dc37be5f636512c#diff-fda2c095ab261c3e019bc5d897986116R3) and this line has been changed into `if @blacklisted_gem`